hubspot/plugin.py

```diff
@@ -1193,50 +1194,38 @@ def hubspot_update_companies(
 def hubspot_search_companies(
     search_criteria: Mapping[str, str],
     return_with_custom_properties: Sequence[str] = (),
-) -> List[HubSpotCompany]:
+    pagination_token: Optional[HubSpotPaginationToken] = None,
+) -> Tuple[List[HubSpotCompany], Optional[HubSpotPaginationToken]]:
     """
-    Search for companies in HubSpot CRM based on various criteria.
-
-    Default properties will always be fetched. However, properties with no values will not be in additional_properties
-    dict. You MUST check whether the property exists in additional_properties before using it.
-
-    Args:
-        search_criteria: A dictionary where keys are the property names (e.g.,
-          "name", "domain") and values are the search values for those properties.
-        return_with_custom_properties: A sequence of custom property names to fetch from found
-            contacts. These will be included in additional_properties if they exist.
+    Search for companies in HubSpot CRM.
 
-    Returns:
-        Sequence[HubSpotCompany]: A list of HubSpotCompany objects matching the search
-            criteria.
+    search_criteria: A dictionary where keys are the property names (e.g.,
+        "name", "domain") and values are the search values for those properties.
+    return_with_custom_properties: A sequence of custom property names to fetch from found
+        contacts. These will be included in additional_properties if they exist.
     """
-    return_with_custom_properties = list(return_with_custom_properties)
-    return_with_custom_properties += (
-        _COMPANY_PROPERTIES_DATETIME
-        + _COMPANY_PROPERTIES_BOOLEAN
-        + _COMPANY_PROPERTIES_NUMBER
-        + _COMPANY_PROPERTIES_STRING
-    )
-    url = "https://api.hubapi.com/crm/v3/objects/companies/search"
-
     # Construct the filters based on the search criteria
     filters = []
     for property_name, value in search_criteria.items():
         filters.append(
             {"propertyName": property_name, "operator": "EQ", "value": value}
```

Comment:
is this a good time to do the search_criteria change you  had in mind a while back? to enable other criteria?

or at least change the API so that it allows for us to do that later, without backwards compat issues.

Reply:
Updated to support almost the full range of filters and filter groups